django/db/models/sql/query.py

```diff
@@ -1768,9 +1766,6 @@ def add_immediate_loading(self, field_names):
         """
         existing, defer = self.deferred_loading
         field_names = set(field_names)
-        if 'pk' in field_names:
-            field_names.remove('pk')
-            field_names.add(self.get_meta().pk.name)
```

Comment:
What happens if you call `defer('id', 'pk')` or `only('id', 'pk')` now? How `field_names` containing both will affect chained `defer()` calls?